js/src/launchAsync.ts

```diff
@@ -63,6 +74,259 @@ let launchResponseResolve: LaunchResponseResolve;
 let launchResponseReject: LaunchReject;
 let launchResponseError: Error;
 
+// Added globals to persist between functions
+let timeoutId: number | null = null;
+let parent: Node = null;
+let iframeContainer: HTMLDivElement = null;
+let iframe: HTMLIFrameElement = null;
+let noscroll: HTMLStyleElement = null;
+let messageHandler: (e: any) => void = null;
+
+enum FuncType { launchAsync, launchWithoutContentAsync }
+
+function _initializeOptions(options?: Options): Options {
+
+    const { allowFullscreen, cookiePolicy, hideExitButton, onExit, parent, timeout, uiZIndex, useWebview } = options || {};
```

Comment:
nit: bothers me a little that these properties are not laid out in the same order as below, make it a bit harder to follow

Reply:
ack